### PR TITLE
Singleton instance must depend on requested token

### PR DESCRIPTION
**Pull request by andrey-yantsen**
Or, at least, you should mention current behaviour in docs.

### CODE REVIEW

File: lib/Mixpanel.php [hunk 1]
```diff
@@ -125,9 +125,9 @@ class Mixpanel extends Base_MixpanelBase {
 
     /**
```
Comment (jbwyme) [inline lib/Mixpanel.php]: Can we update the comment to reflect the new type?

File: lib/Mixpanel.php [hunk 2]
```diff
@@ -149,10 +149,10 @@ public function __construct($token, $options = array()) {
      * @return Mixpanel
      */
     public static function getInstance($token, $options = array()) {
-        if(!isset(self::$_instance)) {
-            self::$_instance = new Mixpanel($token, $options);
+        if(!isset(self::$_instances[$token])) {
```
Comment (jbwyme) [inline lib/Mixpanel.php]: Let's use array_key_exists to check. IIRC, this will throw a notice if the key isn't in there.

**Comment by jbwyme** [comment]
Thanks for the PR! Certainly should be this way. Looks like tests also need to be updated (I'm on mobile but I'm guessing that's why the build is failing)

File: lib/Mixpanel.php (hunk 1 above)
Comment (andrey-yantsen) [inline lib/Mixpanel.php]: Yep, I'll update it in next commit

File: lib/Mixpanel.php (hunk 2 above)
Comment (andrey-yantsen) [inline lib/Mixpanel.php]: Nope, there will be no notice. `isset($a['b']) == array_key_exists('b', $a) && !is_null($a['b'])`

**Comment by andrey-yantsen** [comment]
Oooops, tests failed because of short-array syntax usage, sorry. Fixed in last commit.

**Comment by andrey-yantsen** [comment]
Hi, is there any news on this? :)

**Pull request merged**